File: src/main/scala/s/chain_of_resposibility/like/functional/LimitSupport.scala
package s.chain_of_resposibility.like.functional

import s.chain_of_resposibility.Trouble

object LimitSupport {
  def create(name: String, limit: Int): PartialFunction[Trouble, Unit] = {
    case trouble if trouble.number < limit =>
      MessagePrinter.done(name, trouble)
  }
}

File: src/main/java/j/chain_of_resposibility/like/functional/LimitSupport.java
package j.chain_of_resposibility.like.functional;

public class LimitSupport {
    public static Support create(String name, int limit) {
        return t -> {
            if (t.getNumber() < limit) {
                MessagePrinter.done(name, t);
                return ControlBreak.of(t).withControlBreak(true);
            } else {
                return ControlBreak.of(t);
            }
        };
    }
}
